gramex/cache.py

```diff
@@ -128,6 +128,45 @@ def _template(path, **kwargs):
     return tornado.template.Loader(root, **kwargs).load(name)
 
 
+def excel_reader(path, sheet_name='Sheet1', range=None, name=None, table=None, header=0):
+    # import openpyxl
+    # Read the file using openpyxl
+    # If range (e.g. A10:C30) is specified, return cells in that range
+
+    # If name is specified, return cells in that named range using wb.defined_names
+    # If table is specified, return cells in that named table using ws.tables
+    # Convert into a DataFrame -- using header if specified. header can be int, list of int, None
+    import pandas as pd
+    from openpyxl import load_workbook
+
+    wb = load_workbook(path)
+    data_rows = []
+
+    def get_rows(sheet, range_string):
+        ws = wb[sheet]
+        for row in ws[range_string]:
+            data_rows.append([cell.value for cell in row])
+
+    if range is not None:
+        get_rows(sheet_name, range)
+
+    if name is not None:
+        sheetid = wb.sheetnames.index('sales')
+        range_str = wb.defined_names.get(name, sheetid).attr_text.split('!')[1].replace('$', '')
+        get_rows(sheet_name, range_str)
+
+    if table is not None:
+        get_rows(sheet_name, wb[sheet_name].tables[table].ref)
+
+    header_val = None
+    if header is not None:
+        header_val = list(data_rows[header])
+        del data_rows[header]
```

Comment:
@vinothsm -- I suspect if this handles `header can be list of int` scenario. If you're handling only int and None, you could potential simplify with .. ?

```python
header = None if header is None else list(data_rows.pop(header))
df = pd.DataFrame(data_rows, columns=header)
```
without needing to `del` and intermediate `header_val`.

Reply:
@pratapvardhan  Thanks for reviewing. `header` is int only and agree with your suggestion to simplify like above.

Reply:
Have implemented list of int as a possibility.